feature/state

Ahmed7osny1 opened the pull request.

Ahmed7osny1 committed:
add unit test for create new state in state usecase and add create state helper and add stateUseCase

Ahmed7osny1 committed:
add unit test for creation and Deletion state for (creation state usecase, deletion state usecase) and add create state helper

Abdulrahman-Ragab-01 committed:
add update usecase and its unit test

Ahmed7osny1 committed:
create usecase for all state repository functions and add unit test for these

Ahmed7osny1 committed:
update usecase and unit test
add unit test for id generator and update its function

mosayed01 commented on src/main/kotlin/logic/generateIdHelper/IdGenerator.kt:
```diff
@@ -0,0 +1,11 @@
+
+interface IdGenerator {
+    fun generateId(
+        input: String,
```
```suggestion
        prefix: String,
```

mosayed01 commented on src/main/kotlin/logic/generateIdHelper/generateId.kt:
```diff
@@ -0,0 +1,13 @@
+package com.berlin.logic.generateIdHelper
+
+class DefaultIdGenerator : IdGenerator {
```
```suggestion
class IdGeneratorIpml : IdGenerator {
```

mosayed01 commented on src/main/kotlin/logic/usecase/state/CreationStateUseCase.kt:
```diff
@@ -0,0 +1,31 @@
+
+class CreationStateUseCase(
+    private val stateRepository: StateRepository,
+    private val defaultIdGenerator: DefaultIdGenerator,
```
Why u depend on implementation instead of abstraction?

Ahmed7osny1 replied:
you are right this mistake from me, Depending on an implementation violates (DIP)

Ahmed7osny1 committed:
change naming of id generator class, and these parameters
solve a problem in creation state (change depend class on implementation to abstraction)

mosayed01 commented:
@Ahmed7osny1 please check why there is a compile error

Ahmed7osny1 committed:
resolve error in id generator test
